src/factorizations/polar.jl

```diff
@@ -0,0 +1,53 @@
+using MatrixAlgebraKit:
+  MatrixAlgebraKit,
+  PolarViaSVD,
+  check_input,
+  default_algorithm,
+  left_polar!,
+  right_polar!,
+  svd_compact!
+
+function MatrixAlgebraKit.check_input(::typeof(left_polar!), A::AbstractBlockSparseMatrix)
+  @views for I in eachblockstoredindex(A)
+    m, n = size(A[I])
+    m >= n ||
+      throw(ArgumentError("each input matrix block needs at least as many rows as columns"))
+  end
```

Comment:
In principle this check is also performed when `left_polar` is actually called, so the question is if this error message is more helpful than that one. I'm not saying it isn't, just wanted to pose the question, since I'm fine with either.